libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -265,19 +278,47 @@ def _apply_schema(self) -> None:
         # Index on text_embedding (for similarity search)
         self._session.execute(f"""
             CREATE CUSTOM INDEX IF NOT EXISTS {self._node_table}_text_embedding_index
-            ON {self._keyspace}.{self._node_table}(text_embedding)
+            ON {self.table_name()}(text_embedding)
             USING 'StorageAttachedIndex';
         """)
 
         self._session.execute(f"""
             CREATE CUSTOM INDEX IF NOT EXISTS {self._node_table}_link_from_tags
-            ON {self._keyspace}.{self._node_table}(link_from_tags)
+            ON {self.table_name()}(link_from_tags)
+            USING 'StorageAttachedIndex';
+        """)
+
+        self._session.execute(f"""
+            CREATE CUSTOM INDEX IF NOT EXISTS {self._node_table}_metadata_index
+            ON {self.table_name()}(ENTRIES(metadata_s))
             USING 'StorageAttachedIndex';
         """)
 
     def _concurrent_queries(self) -> ConcurrentQueries:
         return ConcurrentQueries(self._session)
 
+    def _parse_metadata(
```

Comment:
The term "parse" here feels weird. Any details on what this is actually doing? Maybe some comments and/or a better method name would help.

Reply:
function removed via other simplifications.